flink-cdc-common/src/main/java/com/ververica/cdc/common/schema/Schema.java

```diff
@@ -0,0 +1,314 @@
+
+    private final List<String> primaryKeys;
+
+    private final Map<String, String> options;
+
+    private final @Nullable String comment;
+
+    private transient volatile Map<String, Column> nameToColumns;
+
+    private Schema(
+            List<Column> columns,
+            List<String> primaryKeys,
+            Map<String, String> options,
+            @Nullable String comment) {
+        this.columns = columns;
+        this.primaryKeys = primaryKeys;
+        this.options = options;
+        this.comment = comment;
+    }
+
+    /** Returns the number of columns of this schema. */
+    public int getColumnCount() {
+        return columns.size();
+    }
+
+    /** Returns all {@link Column}s of this schema. */
+    public List<Column> getColumns() {
+        return columns;
+    }
+
+    /** Returns all column names. It does not distinguish between different kinds of columns. */
+    public List<String> getColumnNames() {
+        return columns.stream().map(Column::getName).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns all column data types. It does not distinguish between different kinds of columns.
+     */
+    public List<DataType> getColumnDataTypes() {
+        return columns.stream().map(Column::getType).collect(Collectors.toList());
+    }
+
+    /** Returns the primary keys of the table or data collection. */
+    public List<String> primaryKeys() {
+        return primaryKeys;
+    }
+
+    /** Returns the options of the table or data collection. */
+    public Map<String, String> options() {
+        return options;
+    }
+
+    /** Returns the comment of the table or data collection. */
+    public String comment() {
+        return comment;
+    }
+
+    /**
+     * Returns the {@link Column} instance for the given column name.
+     *
+     * @param columnName the name of the column
+     */
+    public Optional<Column> getColumn(String columnName) {
+        initializeNameToColumns();
+        return Optional.ofNullable(nameToColumns.get(columnName));
+    }
+
+    /**
+     * Converts all columns of this schema into a (possibly nested) row data type.
+     *
+     * @see DataTypes#ROW(DataField...)
+     */
+    public DataType toRowDataType() {
+        final DataField[] fields =
+                columns.stream().map(Schema::columnToField).toArray(DataField[]::new);
+        // the row should never be null
+        return ROW(fields).notNull();
+    }
+
+    // -----------------------------------------------------------------------------------
+    private void initializeNameToColumns() {
+        if (nameToColumns == null) {
+            // make the method thread-safe
+            synchronized (this) {
+                // we need to check nullability again here
+                if (nameToColumns == null) {
+                    nameToColumns = new HashMap<>();
+                    for (Column col : columns) {
+                        nameToColumns.put(col.getName(), col);
```

Comment:
It's fine to use `col.getName().toLowerCase()`.

Reply:
The column name is case-sensitive by default in flink, so I do not tend to break this, @zhongqiang